tests/protocol/test_protocol_util.py

```diff
@@ -138,64 +107,66 @@ def test_detect_wire_protocol_no_dhcp(self, WireProtocol, mock_get_lib_dir, _):
         endpoint_file = protocol_util._get_wireserver_endpoint_file_path()
 
         # Test wire protocol when no endpoint file has been written
-        protocol_util._detect_wire_protocol()
+        protocol_util._detect_protocol()
         self.assertEqual(KNOWN_WIRESERVER_IP, protocol_util.get_wireserver_endpoint())
 
-        # Test wire protocol when endpoint was previously detected
-        protocol_util.clear_protocol()
-        with open(endpoint_file, "w+") as endpoint_fd:
-            endpoint_fd.write("baz.qux")
-
-        protocol_util._detect_wire_protocol()
-        self.assertEqual("baz.qux", protocol_util.get_wireserver_endpoint())
-
         # Test wire protocol on dhcp failure
-        protocol_util.clear_protocol()
         protocol_util.osutil.is_dhcp_available.return_value = True
         protocol_util.dhcp_handler.run.side_effect = DhcpError()
 
-        self.assertRaises(ProtocolError, protocol_util._detect_wire_protocol)
+        self.assertRaises(ProtocolError, protocol_util._detect_protocol)
 
-    @patch("azurelinuxagent.common.protocol.util.MetadataProtocol")
     @patch("azurelinuxagent.common.protocol.util.WireProtocol")
-    def test_get_protocol(self, WireProtocol, MetadataProtocol, _):
+    def test_get_protocol(self, WireProtocol, _):
         WireProtocol.return_value = MagicMock()
-        MetadataProtocol.return_value = MagicMock()
 
         protocol_util = get_protocol_util()
         protocol_util.get_wireserver_endpoint = Mock()
         protocol_util._detect_protocol = MagicMock()
-
-        # Test for wire protocol
         protocol_util._save_protocol("WireProtocol")
 
         protocol = protocol_util.get_protocol()
-        self.assertEquals(WireProtocol.return_value, protocol)
-        protocol_util.get_wireserver_endpoint.assert_any_call()
 
-        # Test to ensure protocol persists
-        protocol_util.get_wireserver_endpoint.reset_mock()
-        protocol_util._save_protocol("MetadataProtocol")
-
-        protocol = protocol_util.get_protocol()
         self.assertEquals(WireProtocol.return_value, protocol)
-        protocol_util.get_wireserver_endpoint.assert_not_called()
-
-        # Test for metadata protocol
-        protocol_util.clear_protocol()
-        protocol_util._save_protocol("MetadataProtocol")
-
-        protocol = protocol_util.get_protocol()
-        self.assertEquals(MetadataProtocol.return_value, protocol)
-        protocol_util.get_wireserver_endpoint.assert_not_called()
+        protocol_util.get_wireserver_endpoint.assert_any_call()
 
-        # Test for unknown protocol
-        protocol_util.clear_protocol()
-        protocol_util._save_protocol("Not_a_Protocol")
-        protocol_util._detect_protocol.side_effect = NotImplementedError()
+    @patch("os.path.isfile")
```

Comment:
this is an excessive use of mocks that make those tests too fragile. let's sync offline